dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -982,21 +982,19 @@ std::vector<BlobFileId> BlobStore::getGCStats()
             // Avoid divide by zero
             if (right_margin == 0)
             {
-                if (unlikely(stat->sm_valid_rate != 0))
+                // Ignore the BlobFile if it is already truncated
+                if (stat->sm_total_size != 0)
                 {
-                    throw Exception(fmt::format("Current blob is empty, but valid rate is not 0. [blob_id={}][valid_size={}][valid_rate={}]",
-                                                stat->id,
-                                                stat->sm_valid_size,
-                                                stat->sm_valid_rate));
+                    RUNTIME_CHECK_MSG(stat->sm_valid_rate == 0, "Current blob is empty, but valid rate is not 0. [blob_id={}][valid_size={}][valid_rate={}]", stat->id, stat->sm_valid_size, stat->sm_valid_rate);
```

Comment:
`sm_valid_rate` is a double, comparing it directly to 0 may fail. Should we compare stat->sm_valid_rate is closed to 0.0 by a epsilon?

Reply:
Use `sm_valid_size` instead.